tests/kernels/test_causal_conv1d.py

```diff
@@ -344,17 +332,25 @@ def test_causal_conv1d_update_with_batch_gather(dim, width, seqlen, has_bias,
 @pytest.mark.parametrize('seqlen',
                          [8, 16, 32, 64, 128, 256, 512, 784, 1024, 2048, 4096])
 @pytest.mark.parametrize('dim', [64, 4096])
-def test_causal_conv1d_varlen(dim, seqlen, width, has_bias, silu_activation,
-                              itype):
+@pytest.mark.parametrize('with_padding', [True, False])
+def test_causal_conv1d_varlen(with_padding, dim, seqlen, width, has_bias,
+                              silu_activation, itype):
     device = "cuda"
     rtol, atol = (3e-4, 1e-3) if itype == torch.float32 else (3e-3, 5e-3)
     if itype == torch.bfloat16:
         rtol, atol = 1e-2, 5e-2
     # set seed
     seed_everything(0)
-    batch = 1
     seqlens = []
-    nsplits = 3
+    batch_size = 4
+    if seqlen < 10:
+        if with_padding:
+            pytest.skip()
+        batch_size = 1
+    padding = 3 if with_padding else 0
```

Comment:
This seems a little convoluted. Not sure I follow what is happening here - possible to simplify this code?

Reply:
Addressed :)